include/nuttx/mmcsd.h

```diff
@@ -49,6 +54,49 @@
  * Public Types
  ****************************************************************************/
 
+struct mmc_ioc_cmd
+{
+  /* Direction of data: nonzero = write, zero = read.
+   * Bit 31 selects 'Reliable Write' for RPMB.
+   */
+
+  int write_flag;
+
+  /* Application-specific command.  true = precede with CMD55 */
+
+  int is_acmd;
+
+  uint32_t opcode;
+  uint32_t arg;
+  uint32_t response[4];  /* CMD response */
+  unsigned int flags;
+  unsigned int blksz;
+  unsigned int blocks;
+
+  /* For 64-bit machines, the next member, ``uint64_t data_ptr``, wants to
+   * be 8-byte aligned.  Make sure this struct is the same size when
+   * built for 32-bit.
+   */
+
+  uint32_t pad;
+
+  /* DAT buffer */
+
+  uint64_t data_ptr;
```

Comment:
Where this requirement comes from? I do not see that address of `data_ptr` is taken anywhere

Reply:
It's used here:
https://github.com/apache/nuttx/pull/8256/files#diff-161d00b0d61f441522bd0e1fbdb8c154db2c56fcdd69d484d0268c50b4788373R3070
https://github.com/apache/nuttx/pull/8256/files#diff-161d00b0d61f441522bd0e1fbdb8c154db2c56fcdd69d484d0268c50b4788373R3070